src/App.tsx

```diff
@@ -1,223 +1,108 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+// eslint-disable-next-line
+import React, { useState, useMemo, useContext, useEffect, useRef } from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/FilterType';
+import { Error } from './types/Error';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+// eslint-disable-next-line
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<Error>({
+    hasError: false,
+    hasMessage: '',
+  });
```

Comment:
Do you need `hasError` property? If `hasMessage` !== '' it has error. You can replace your state value with simple string.